### PR TITLE
Refactor/fragments

### PR DESCRIPTION
**Pull request by Bwonner**
Adopt java methods to kotlin
Refactor fragments
Migrate to ViewPager2
Add api

### CODE REVIEW

File: app/src/main/java/com/innowisegroup/datetimepicker/DateExt.kt [hunk 1]
```diff
@@ -1,19 +1,19 @@
 package com.innowisegroup.datetimepicker
 
 internal fun LocalTime.formatTime(): String {
-    val hour = this.getHour()
+    val hour = hour
```
Comment (dmorskoi) [inline app/src/main/java/com/innowisegroup/datetimepicker/DateExt.kt]: Do we need this val. I think we can inline this value in "if" statement

File: app/src/main/java/com/innowisegroup/datetimepicker/DateExt.kt [hunk 2]
```diff
@@ -1,19 +1,19 @@
...
     val correctHour = if (hour < 10) "0$hour" else "$hour"
-    val minute = this.getMinute()
+    val minute = minute
```
Comment (dmorskoi) [inline app/src/main/java/com/innowisegroup/datetimepicker/DateExt.kt]: Do we need this val. I think we can inline this value in "if" statement

File: app/src/main/java/com/innowisegroup/datetimepicker/DateExt.kt [hunk 3]
```diff
@@ -1,19 +1,19 @@
...
     val correctMinute = if (minute < 10) "0$minute" else "$minute"
     return "$correctHour:$correctMinute"
 }
 
 internal fun LocalDate.formatDate(): String {
-    val day = this.getDay()
+    val day = day
```
Comment (dmorskoi) [inline app/src/main/java/com/innowisegroup/datetimepicker/DateExt.kt]: the same regarding val

File: app/src/main/java/com/innowisegroup/datetimepicker/DateExt.kt [hunk 4]
```diff
@@ -1,19 +1,19 @@
...
     val correctDay = if (day < 10) "0$day" else "$day"
-    val month = this.getMonth()
+    val month = month
```
Comment (dmorskoi) [inline app/src/main/java/com/innowisegroup/datetimepicker/DateExt.kt]: the same regarding val

File: app/src/main/java/com/innowisegroup/datetimepicker/DateExt.kt [hunk 5]
```diff
@@ -1,19 +1,19 @@
...
     val correctMonth = if (month < 10) "0$month" else "$month"
-    val year = this.getYear()
+    val year = year
```
Comment (dmorskoi) [inline app/src/main/java/com/innowisegroup/datetimepicker/DateExt.kt]: the same regarding val

File: app/src/main/java/com/innowisegroup/datetimepicker/DateTimePickerDialog.kt [hunk 6]
```diff
@@ -9,50 +9,62 @@ import android.widget.Button
...
-    init {
-        this.initialLocalDateTime = initialLocalDateTime
-                ?: LocalDateTime().now()
-    }
+    private var minLocalDateTime: LocalDateTime? = null
+    private var maxLocalDateTime: LocalDateTime? = null
+    private var wrapSelectionWheel: Boolean = false
+    private var fragmentToCreate: FragmentToCreate = FragmentToCreate.BOTH
+
+    private var currentTimeStub = "99:99"
```
Comment (dmorskoi) [inline app/src/main/java/com/innowisegroup/datetimepicker/DateTimePickerDialog.kt]: is this a real time stub??)

File: app/src/main/java/com/innowisegroup/datetimepicker/DateTimePickerDialog.kt [hunk 7]
```diff
@@ -9,50 +9,62 @@ import android.widget.Button
...
+    private var currentDateStub = "99-99-99"
```
Comment (dmorskoi) [inline app/src/main/java/com/innowisegroup/datetimepicker/DateTimePickerDialog.kt]: The same here

File: app/src/main/java/com/innowisegroup/datetimepicker/PagerAdapter.kt [hunk 8]
```diff
@@ -1,16 +1,22 @@
...
-open class PagerAdapter(fm: FragmentManager) : FragmentPagerAdapter(fm) {
+class PagerAdapter(fragmentActivity: FragmentActivity, private val list: List<Fragment>) :
+    FragmentStateAdapter(fragmentActivity) {
 
-    private val mFragmentList: MutableList<Fragment> = ArrayList()
+    override fun getItemCount(): Int = list.size
 
-    override fun getItem(position: Int): Fragment = mFragmentList[position]
+    override fun createFragment(position: Int): Fragment {
+        return list[position]
```
Comment (AndrKriv) [inline app/src/main/java/com/innowisegroup/datetimepicker/PagerAdapter.kt]: refactor return to "="

File: app/src/main/java/com/innowisegroup/datetimepicker/PagerAdapter.kt [hunk 9]
```diff
@@ -1,16 +1,22 @@
...
+    }
 
-    override fun getCount(): Int = mFragmentList.size
-
-    fun addFragment(fragment: Fragment) = mFragmentList.add(fragment)
+    override fun getItemViewType(position: Int): Int {
+        return when(list[position]) {
```
Comment (AndrKriv) [inline app/src/main/java/com/innowisegroup/datetimepicker/PagerAdapter.kt]: refactor return to "="

File: app/src/main/java/com/innowisegroup/datetimepicker/DateTimePickerDialog.kt [hunk 10]
```diff
@@ -9,50 +9,62 @@ import android.widget.Button
...
 
     override fun onCreateView(
-            inflater: LayoutInflater,
-            container: ViewGroup?,
-            savedInstanceState: Bundle?
```
Comment (dmorskoi) [inline app/src/main/java/com/innowisegroup/datetimepicker/DateTimePickerDialog.kt:47]: Seems like you and Andrew have different formatting? Make a sync and check if you have the same code formatting to prevent unnecessary code changes

File: app/src/main/java/com/innowisegroup/datetimepicker/DateTimePickerDialog.kt [hunk 11]
```diff
@@ -9,50 +9,62 @@ import android.widget.Button
...
+                getSerializable(FRAGMENT_TO_CREATE) as? FragmentToCreate ?: FragmentToCreate.BOTH
+        }
+
         isCancelable = false
         dialog?.window?.setBackgroundDrawable(
-                ResourcesCompat.getDrawable(resources,
-                        R.drawable.background_datetimepicker_dialog,
-                        null
-                )
+            ResourcesCompat.getDrawable(
```
Comment (dmorskoi) [inline app/src/main/java/com/innowisegroup/datetimepicker/DateTimePickerDialog.kt]: Code formatting!

File: app/src/main/java/com/innowisegroup/datetimepicker/FragmentToCreate.kt [hunk 12]
```diff
@@ -0,0 +1,5 @@
+package com.innowisegroup.datetimepicker
+
+enum class FragmentToCreate {
+    ONLY_TIME, ONLY_DATE, BOTH
```
Comment (dmorskoi) [inline app/src/main/java/com/innowisegroup/datetimepicker/FragmentToCreate.kt]: Let's rename to - PickerType 
and -  TIME_ONLY, DATE_ONLY, DATE_TIME

File: app/src/main/java/com/innowisegroup/datetimepicker/DateTimePickerDialog.kt [hunk 13]
```diff
@@ -65,50 +77,69 @@ class DateTimePickerDialog(
...
+            )
+            FragmentToCreate.ONLY_DATE -> listOf(
+                FragmentDatePicker.newInstance(
                     initialLocalDateTime?.toLocalDate(),
-                    minLocalDateTime?.toLocalDate(),
-                    maxLocalDateTime?.toLocalDate(),
-                    object : RefreshCallback {
-                        override fun refresh() = this@DateTimePickerDialog.refresh()
-                    },
+                    if (minLocalDateTime == null) null else minLocalDateTime?.toLocalDate(),
```
Comment (dmorskoi) [inline app/src/main/java/com/innowisegroup/datetimepicker/DateTimePickerDialog.kt]: let's think how we can avoid (== null) null else

File: app/src/main/java/com/innowisegroup/datetimepicker/DateTimePickerDialog.kt [hunk 14]
```diff
@@ -180,7 +207,42 @@ class DateTimePickerDialog(
...
     }
 
-    private companion object {
-        const val DIALOG_TAG = "date_time_picker_dialog"
+    companion object {
+        const val UPDATE_TIME_TAB_TITLE_REQUEST_KEY = "updateTimeTabTitleKey"
+        const val UPDATE_DATE_TAB_TITLE_REQUEST_KEY = "updateDateTabTitleKey"
+
+        const val UPDATE_TIME_TAB_TITLE_KEY = "updateTimeTabTitleKey"
+        const val UPDATE_DATE_TAB_TITLE_KEY = "updateDateTabTitleKey"
```
Comment (dmorskoi) [inline app/src/main/java/com/innowisegroup/datetimepicker/DateTimePickerDialog.kt]: Should these constants really be public?

File: app/src/main/java/com/innowisegroup/datetimepicker/MainActivity.kt [hunk 15]
```diff
@@ -10,13 +10,14 @@ class MainActivity : AppCompatActivity() {
...
-                wrapSelectionWheel = false,
-                withOnlyDatePicker = false,
-                withOnlyTimePicker = false)
+        val dateTimePickerDialog = DateTimePickerDialog.newInstance(
+            initialLocalDateTime = null,
+            initialLocalTime = null,
+            minLocalDateTime = null,
+            maxLocalDateTime = null,
+            wrapSelectionWheel = false,
+            fragmentToCreate = FragmentToCreate.BOTH
```
Comment (dmorskoi) [inline app/src/main/java/com/innowisegroup/datetimepicker/MainActivity.kt]: Such an API (null, null, null.....) is not what the developer wants to see when creating picker. Let's think how we can improve it and make more friendly to call.

File: app/src/main/java/com/innowisegroup/datetimepicker/PagerAdapter.kt [hunk 16]
```diff
@@ -1,16 +1,22 @@
 package com.innowisegroup.datetimepicker
 
 import androidx.fragment.app.Fragment
-import androidx.fragment.app.FragmentManager
-import androidx.fragment.app.FragmentPagerAdapter
+import androidx.fragment.app.FragmentActivity
+import androidx.viewpager2.adapter.FragmentStateAdapter
 
-open class PagerAdapter(fm: FragmentManager) : FragmentPagerAdapter(fm) {
+class PagerAdapter(fragmentActivity: FragmentActivity, private val list: List<Fragment>) :
```
Comment (dmorskoi) [inline app/src/main/java/com/innowisegroup/datetimepicker/PagerAdapter.kt]: move properties in constructor to separate lines

File: app/src/main/java/com/innowisegroup/datetimepicker/DateTimePickerDialog.kt [hunk 17]
```diff
@@ -180,7 +207,42 @@ class DateTimePickerDialog(
...
+        @JvmStatic
+        fun newInstance(
+            initialLocalDateTime: LocalDateTime?,
+            initialLocalTime: LocalTime?,
+            minLocalDateTime: LocalDateTime?,
+            maxLocalDateTime: LocalDateTime?,
+            fragmentToCreate: FragmentToCreate,
+            wrapSelectionWheel: Boolean
+        ): DateTimePickerDialog {
+            val args = Bundle().apply {
```
Comment (dmorskoi) [inline app/src/main/java/com/innowisegroup/datetimepicker/DateTimePickerDialog.kt]: let's use bundleOf() function

File: app/src/main/java/com/innowisegroup/datetimepicker/FragmentDatePicker.kt [hunk 18]
```diff
@@ -62,88 +50,101 @@ class FragmentDatePicker : Fragment() {
...
+        private const val WRAP_SELECTION_BOOLEAN = "wrapSelectionWheel"
+
+        @JvmStatic
+        fun newInstance(
+            localDate: LocalDate?,
+            minLocalDate: LocalDate?,
+            maxLocalDate: LocalDate?,
+            wrapSelectionWheel: Boolean
+        ): FragmentDatePicker {
+            val args = Bundle().apply {
```
Comment (dmorskoi) [inline app/src/main/java/com/innowisegroup/datetimepicker/FragmentDatePicker.kt]: let's use bundleOf() function

File: app/src/main/java/com/innowisegroup/datetimepicker/FragmentTimePicker.kt [hunk 19]
```diff
@@ -43,53 +39,68 @@ class FragmentTimePicker : Fragment() {
...
+        private const val MIN_MINUTE = 0
+        private const val MAX_MINUTE = 59
+
+        private const val LOCAL_TIME = "localDate"
+
+        @JvmStatic
+        fun newInstance(
+            localTime: LocalTime?,
+        ): FragmentTimePicker {
+            val args = Bundle().apply {
```
Comment (dmorskoi) [inline app/src/main/java/com/innowisegroup/datetimepicker/FragmentTimePicker.kt]: let's use bundleOf() function

File: app/src/main/java/com/innowisegroup/datetimepicker/FragmentTimePicker.kt [hunk 20]
```diff
@@ -43,53 +39,68 @@ class FragmentTimePicker : Fragment() {
...
             refreshTimeValue(
-                    localTime?.withMinute(newVal)!!
+                localTime?.withMinute(newVal)!!
             )
         }
         return view
     }
 
     private fun refreshTimeValue(newValue: LocalTime) {
-        localTime = newValue
```
Comment (AndrKriv) [inline app/src/main/java/com/innowisegroup/datetimepicker/FragmentTimePicker.kt:85]: return this line. now you have the incorrect value in the tab when you change the value in the picker

File: app/src/main/java/com/innowisegroup/datetimepicker/DateTimePickerDialog.kt (hunk 6 above)
Comment (Bwonner) [reply app/src/main/java/com/innowisegroup/datetimepicker/DateTimePickerDialog.kt]: Just an eye catching number

File: app/src/main/java/com/innowisegroup/datetimepicker/FragmentDatePicker.kt [hunk 21]
```diff
@@ -62,88 +50,101 @@ class FragmentDatePicker : Fragment() {
...
-        )
         return view
     }
 
     private fun refreshDateValue(newValue: LocalDate) {
-        if (day == null) {
-            return
-        }
-        localDate = newValue
-        day?.maxValue = localDate?.lengthOfMonth()!!
```
Comment (AndrKriv) [inline app/src/main/java/com/innowisegroup/datetimepicker/FragmentDatePicker.kt:138]: Why did you delete these lines? Now you have the incorrect value in the tab. These lines helps to set the correct value to the tab and set how many days in a month. Return these lines

File: app/src/main/java/com/innowisegroup/datetimepicker/DateTimePickerDialog.kt (hunk 17 above)
Comment (Bwonner) [reply app/src/main/java/com/innowisegroup/datetimepicker/DateTimePickerDialog.kt]: image

File: app/src/main/java/com/innowisegroup/datetimepicker/DateTimePickerDialog.kt (hunk 14 above)
Comment (Bwonner) [reply app/src/main/java/com/innowisegroup/datetimepicker/DateTimePickerDialog.kt]: Its a key for all the fragments

File: app/src/main/java/com/innowisegroup/datetimepicker/FragmentTimePicker.kt [hunk 22]
```diff
@@ -43,53 +39,68 @@ class FragmentTimePicker : Fragment() {
...
-        const val MAX_HOUR = 23
-        const val MIN_MINUTE = 0
-        const val MAX_MINUTE = 59
+    companion object {
+        private const val MIN_HOUR = 0
+        private const val MAX_HOUR = 23
+        private const val MIN_MINUTE = 0
+        private const val MAX_MINUTE = 59
+
+        private const val LOCAL_TIME = "localDate"
```
Comment (AndrKriv) [inline app/src/main/java/com/innowisegroup/datetimepicker/FragmentTimePicker.kt]: change to "localTime"

File: app/src/main/java/com/innowisegroup/datetimepicker/DateTimePickerDialog.kt [hunk 23]
```diff
@@ -81,39 +73,41 @@ class DateTimePickerDialog : DialogFragment() {
...
                     initialLocalDateTime?.toLocalDate(),
-                    if (minLocalDateTime == null) null else minLocalDateTime?.toLocalDate(),
-                    if (maxLocalDateTime == null) null else maxLocalDateTime?.toLocalDate(),
+                    if (minLocalDateTime == null) MIN_DEFAULT_DATE_TIME.toLocalDate() else minLocalDateTime?.toLocalDate(),
+                    if (maxLocalDateTime == null) MAX_DEFAULT_DATE_TIME.toLocalDate() else maxLocalDateTime?.toLocalDate(),
                     wrapSelectionWheel
                 )
             )
-            FragmentToCreate.BOTH -> {
+            PickerType.BOTH -> {
```
Comment (dmorskoi) [inline app/src/main/java/com/innowisegroup/datetimepicker/DateTimePickerDialog.kt]: also change here as I commented - PickerType.DATE_TIME. 
It is not clear if I see in code PickerType.BOTH what BOTH means. DATE_TIME more obvious.

File: app/src/main/java/com/innowisegroup/datetimepicker/DateTimePickerDialog.kt [hunk 24]
```diff
@@ -180,7 +200,44 @@ class DateTimePickerDialog(
...
+                putSerializable(INITIAL_LOCAL_DATE_TIME, initialLocalDateTime)
+                putSerializable(MIN_LOCAL_DATE_TIME, minLocalDateTime)
+                putSerializable(MAX_LOCAL_DATE_TIME, maxLocalDateTime)
+                putSerializable(FRAGMENT_TO_CREATE, pickerType)
+                putBoolean(WRAP_SELECTION_WHEEL, wrapSelectionWheel)
+            }
+            val fragment = DateTimePickerDialog()
+            fragment.arguments = args
+            return fragment
+        }
```
Comment (dmorskoi) [inline app/src/main/java/com/innowisegroup/datetimepicker/DateTimePickerDialog.kt]: Let's make it in Kotlin consice way

            DateTimePickerDialog().apply {
                arguments = Bundle().apply {
                    putSerializable(INITIAL_LOCAL_DATE_TIME, initialLocalDateTime)
                    putSerializable(MIN_LOCAL_DATE_TIME, minLocalDateTime)
                    putSerializable(MAX_LOCAL_DATE_TIME, maxLocalDateTime)
                    putSerializable(FRAGMENT_TO_CREATE, pickerType)
                    putBoolean(WRAP_SELECTION_WHEEL, wrapSelectionWheel)
                }
            }

Comment (dmorskoi) [reply app/src/main/java/com/innowisegroup/datetimepicker/DateTimePickerDialog.kt]: Make the same in other places where "newInstance" method is called

File: app/src/main/java/com/innowisegroup/datetimepicker/DateTimePickerDialog.kt (hunk 17 above)
Comment (dmorskoi) [reply app/src/main/java/com/innowisegroup/datetimepicker/DateTimePickerDialog.kt]: Yeah. It comes with separate dependency. Then let's skip it.

File: app/src/main/java/com/innowisegroup/datetimepicker/MainActivity.kt [hunk 25]
```diff
@@ -12,11 +12,10 @@ class MainActivity : AppCompatActivity() {
 
         val dateTimePickerDialog = DateTimePickerDialog.newInstance(
             initialLocalDateTime = null,
-            initialLocalTime = null,
             minLocalDateTime = null,
             maxLocalDateTime = null,
             wrapSelectionWheel = false,
-            fragmentToCreate = FragmentToCreate.BOTH
+            pickerType = PickerType.BOTH
```
Comment (dmorskoi) [inline app/src/main/java/com/innowisegroup/datetimepicker/MainActivity.kt]: I don't like such instance creation. Can we make it better?

File: app/src/main/java/com/innowisegroup/datetimepicker/DateTimePickerDialog.kt [hunk 26]
```diff
@@ -201,43 +234,78 @@ class DateTimePickerDialog : DialogFragment() {
...
+        private val MIN_DEFAULT_DATE_TIME = LocalDateTime(MIN_DEFAULT_DATE, MIN_DEFAULT_TIME)
+        private val MAX_DEFAULT_DATE_TIME = LocalDateTime(MAX_DEFAULT_DATE, MAX_DEFAULT_TIME)
+
+        @JvmStatic
+        @JvmOverloads
+        fun createTimePickerDialog(
+            initialLocalTime: LocalTime = LocalTime.now(),
+            minLocalTime: LocalTime = MIN_DEFAULT_TIME,
+            maxLocalTime: LocalTime = MAX_DEFAULT_TIME,
+            pickerType: PickerType = PickerType.TIME_ONLY,
```
Comment (AndrKriv) [inline app/src/main/java/com/innowisegroup/datetimepicker/DateTimePickerDialog.kt]: Remove this line pickerType: PickerType = PickerType.TIME_ONLY from constructor. You should not put a value in the constructor, which is explicitly defined for the picker type

File: app/src/main/java/com/innowisegroup/datetimepicker/FragmentDatePicker.kt [hunk 27]
```diff
@@ -62,71 +46,64 @@ class FragmentDatePicker : Fragment() {
...
             )
         }
         day?.wrapSelectorWheel = wrapSelectionWheel
-        day?.setOnValueChangedListener { _: NumberPicker?, _: Int, newVal: Int ->
+        day?.setOnValueChangedListener { _, _, newVal: Int ->
             refreshDateValue(
-                    localDate?.withDayOfMonth(newVal)!!
+                localDate?.withDayOfMonth(newVal)!!
             )
         }
```
Comment (dmorskoi) [inline app/src/main/java/com/innowisegroup/datetimepicker/FragmentDatePicker.kt]: Let's avoid duplications in code

        day?.run {
            minValue = DAY_MIN_VALUE
            maxValue = localDate?.lengthOfMonth()!!
            value = localDate?.getDay()!!
            setDividerColor(
                day,
                ResourcesCompat.getDrawable(resources, R.drawable.number_picker_divider_color, null)
            )
            setFormatter { i: Int ->
                String.format(
                    Locale.getDefault(),
                    "%02d",
                    i
                )
            }
            wrapSelectorWheel = wrapSelectionWheel
            setOnValueChangedListener { _, _, newVal: Int ->
                refreshDateValue(
                    localDate?.withDayOfMonth(newVal)!!
                )
            }
        }

File: app/src/main/java/com/innowisegroup/datetimepicker/FragmentDatePicker.kt [hunk 28]
```diff
@@ -62,71 +46,64 @@ class FragmentDatePicker : Fragment() {
...
 
         month?.minValue = MONTH_MIN_VALUE
         month?.maxValue = MONTH_MAX_VALUE
         month?.value = localDate?.getMonth()!!
         month?.setDividerColor(
```
Comment (dmorskoi) [inline app/src/main/java/com/innowisegroup/datetimepicker/FragmentDatePicker.kt]: the same here

File: app/src/main/java/com/innowisegroup/datetimepicker/FragmentDatePicker.kt [hunk 29]
```diff
@@ -62,71 +46,64 @@ class FragmentDatePicker : Fragment() {
...
         month?.wrapSelectorWheel = wrapSelectionWheel
-        month?.setOnValueChangedListener { _: NumberPicker?, _: Int, newVal: Int ->
+        month?.setOnValueChangedListener { _, _, newVal: Int ->
             refreshDateValue(
-                    localDate?.withMonth(newVal)!!
+                localDate?.withMonth(newVal)!!
             )
         }
 
         year?.minValue = minLocalDate?.getMinYear()!!
```
Comment (dmorskoi) [inline app/src/main/java/com/innowisegroup/datetimepicker/FragmentDatePicker.kt]: and here

File: app/src/main/java/com/innowisegroup/datetimepicker/FragmentTimePicker.kt [hunk 30]
```diff
@@ -43,53 +39,57 @@ class FragmentTimePicker : Fragment() {
         mHours?.wrapSelectorWheel = true
```
Comment (dmorskoi) [inline app/src/main/java/com/innowisegroup/datetimepicker/FragmentTimePicker.kt]: the same about duplications

File: app/src/main/java/com/innowisegroup/datetimepicker/PickerDialog.kt [hunk 31]
```diff
@@ -0,0 +1,324 @@
...
+    }
+
+    companion object {
+        private const val DIALOG_TAG = "date_time_picker_dialog"
+
+        internal const val UPDATE_TIME_TAB_TITLE_REQUEST_KEY = "updateTimeTabTitleKey"
+        internal const val UPDATE_DATE_TAB_TITLE_REQUEST_KEY = "updateDateTabTitleKey"
+
+        internal const val UPDATE_TIME_TAB_TITLE_KEY = "updateTimeTabTitleKey"
+        internal const val UPDATE_DATE_TAB_TITLE_KEY = "updateDateTabTitleKey"
```
Comment (dmorskoi) [inline app/src/main/java/com/innowisegroup/datetimepicker/PickerDialog.kt]: You have identical string values in 2 different constants. 
UPDATE_TIME_TAB_TITLE_KEY = "updateTimeTabTitleKey"
UPDATE_TIME_TAB_TITLE_REQUEST_KEY = "updateTimeTabTitleKey"

To prevent potential bugs alway use different values for constants

File: app/src/main/java/com/innowisegroup/datetimepicker/PickerDialog.kt [hunk 32]
```diff
@@ -0,0 +1,324 @@
...
+                LocalDateTime.of(maxLocalTime),
+                PickerType.TIME_ONLY,
+                wrapSelectionWheel
+            )
+
+        @JvmStatic
+        @JvmOverloads
+        fun createDatePickerDialog(
+            initialLocalDate: LocalDate = LocalDate.now(),
+            minLocalDateTime: LocalDate = MIN_DEFAULT_DATE,
```
Comment (dmorskoi) [inline app/src/main/java/com/innowisegroup/datetimepicker/PickerDialog.kt]: minLocalDate

File: app/src/main/java/com/innowisegroup/datetimepicker/PickerDialog.kt [hunk 33]
```diff
@@ -0,0 +1,324 @@
...
+            maxLocalDateTime: LocalDate = MAX_DEFAULT_DATE,
```
Comment (dmorskoi) [inline app/src/main/java/com/innowisegroup/datetimepicker/PickerDialog.kt]: maxLocalDate

File: app/src/main/java/com/innowisegroup/datetimepicker/PickerDialog.kt [hunk 34]
```diff
@@ -0,0 +1,324 @@
...
+                LocalDateTime.of(minLocalDateTime),
+                LocalDateTime.of(maxLocalDateTime),
+                PickerType.DATE_ONLY,
+                wrapSelectionWheel
+            )
+
+        @JvmStatic
+        @JvmOverloads
+        fun createDateTimePickerDialog(
+            initialLocalDate: LocalDateTime = LocalDateTime.now(),
```
Comment (dmorskoi) [inline app/src/main/java/com/innowisegroup/datetimepicker/PickerDialog.kt]: initialLocalDateTime

File: app/src/main/java/com/innowisegroup/datetimepicker/PickerDialog.kt [hunk 35]
```diff
@@ -0,0 +1,324 @@
...
+            maxLocalDateTime: LocalDateTime,
+            pickerType: PickerType,
+            wrapSelectionWheel: Boolean
+        ): PickerDialog =
+            PickerDialog().apply {
+                arguments = Bundle().apply {
+                    putSerializable(INITIAL_LOCAL_DATE_TIME, initialLocalDateTime)
+                    putSerializable(MIN_LOCAL_DATE_TIME, minLocalDateTime)
+                    putSerializable(MAX_LOCAL_DATE_TIME, maxLocalDateTime)
+                    putSerializable(FRAGMENT_TO_CREATE, pickerType)
```
Comment (dmorskoi) [inline app/src/main/java/com/innowisegroup/datetimepicker/PickerDialog.kt]: FRAGMENT_TO_CREATE - lets rename it too - PICKER_TYPE

File: app/src/main/java/com/innowisegroup/datetimepicker/MainActivity.kt [hunk 36]
```diff
@@ -10,16 +10,11 @@ class MainActivity : AppCompatActivity() {
...
-                null,
-                wrapSelectionWheel = false,
-                withOnlyDatePicker = false,
-                withOnlyTimePicker = false)
+        val pickerDialog = PickerDialog.createDateTimePickerDialog()
+
         val btn = findViewById<Button>(R.id.btn)
         btn.setOnClickListener {
-            dateTimePickerDialog.showDialog(supportFragmentManager)
+            pickerDialog.showDialog(supportFragmentManager)
```
Comment (dmorskoi) [inline app/src/main/java/com/innowisegroup/datetimepicker/MainActivity.kt]: So is it "nice to call" API - PickerDialog.createDateTimePickerDialog().showDialog(supportFragmentManager)
Hmm...Every word ends with ...dialog

1. Probably, let's rename it to ReelPicker and function calls too (See below)
2. also rename FragmentDatePicker -> DatePickerFragment and FragmentTimePicker -> TimePickerFragment (in separate commit)

ReelPicker.createDateDialog().show()
ReelPicker.createTimeDialog().show()
ReelPicker.createDateTimeDialog().show()

**Review by AndrKriv — APPROVED** [review]

**Review by dmorskoi — APPROVED** [review]
